src/omv/modules/py_ml.c

```diff
@@ -40,20 +40,53 @@ static size_t py_ml_tuple_sum(mp_obj_tuple_t *o) {
 static void py_ml_process_input(py_ml_model_obj_t *model, mp_obj_t arg) {
     mp_obj_list_t *input_list = MP_OBJ_TO_PTR(arg);
 
-    for (size_t i = 0; i < model->inputs_size; i++) {
+    for (size_t i = 0; i < model->input_shape->len; i++) {
         void *input_buffer = ml_backend_get_input(model, i);
         size_t input_size = py_ml_tuple_sum(MP_OBJ_TO_PTR(model->input_shape->items[i]));
         mp_obj_tuple_t *input_shape = MP_OBJ_TO_PTR(model->input_shape->items[i]);
+        float input_scale = 1.0f / mp_obj_get_float(model->input_scale->items[i]);
+        int input_zero_point = mp_obj_get_int(model->input_zero_point->items[i]);
+        int input_dtype = mp_obj_get_int(model->input_dtype->items[i]);
         mp_obj_t input_arg = input_list->items[i];
 
         if (mp_obj_is_callable(input_arg)) {
             // Input is a callable. Call the object and pass the tensor buffer and dtype.
             mp_obj_t fargs[3] = {
                 mp_obj_new_bytearray_by_ref(input_size, input_buffer),
                 MP_OBJ_FROM_PTR(input_shape),
-                mp_obj_new_int(model->input_dtype)
+                mp_obj_new_int(input_dtype)
             };
             mp_call_function_n_kw(input_arg, 3, 0, fargs);
+        } else if (MP_OBJ_IS_TYPE(input_arg, &mp_type_tuple) || MP_OBJ_IS_TYPE(input_arg, &mp_type_list)) {
+            size_t len;
+            mp_obj_t *items;
+            mp_obj_get_array(input_arg, &len, &items);
+
+            if (len != py_ml_tuple_sum(input_shape)) {
+                mp_raise_msg(&mp_type_ValueError, MP_ERROR_TEXT("Input shape does not match the model input shape"));
+            }
+
+            if (input_dtype == 'f') {
+                float *model_input_float = (float *) input_buffer;
+                for (size_t i = 0; i < len; i++) {
+                    model_input_float[i] = mp_obj_get_float(items[i]);
+                }
+            } else if (input_dtype == 'b') {
+                int8_t *model_input_8 = (int8_t *) input_buffer;
+                for (size_t i = 0; i < len; i++) {
+                    model_input_8[i] = (int8_t) ((mp_obj_get_float(items[i]) * input_scale) + input_zero_point);
+                }
+            } else if (input_dtype == 'B') {
+                int8_t *model_input_8 = (int8_t *) input_buffer;
+                for (size_t i = 0; i < len; i++) {
+                    model_input_8[i] = (uint8_t) ((mp_obj_get_float(items[i]) * input_scale) + input_zero_point);
+                }
+            } else {
+                int16_t *model_input_16 = (int16_t *) input_buffer;
+                for (size_t i = 0; i < len; i++) {
+                    model_input_16[i] = (int16_t) ((mp_obj_get_float(items[i]) * input_scale) + input_zero_point);
+                }
+            }
```

Comment:
```suggestion
```
Please remove this. ndarray or list/tuple not both as discussed. A better way to support this now, is creating ndarrays from lists or tuples in `ml/model.py`

Reply:
Done